src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -1360,64 +1360,59 @@ void Instance::OnFailSafeTimerExpired()
     }
 }
 
-CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)
+CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster,
+                                               DataModel::ListBuilder<DataModel::AcceptedCommandEntry> & builder)
 {
     using namespace Clusters::NetworkCommissioning::Commands;
 
-    if (mFeatureFlags.Has(Feature::kThreadNetworkInterface))
-    {
-        for (auto && cmd : {
-                 ScanNetworks::Id,
-                 AddOrUpdateThreadNetwork::Id,
-                 RemoveNetwork::Id,
-                 ConnectNetwork::Id,
-                 ReorderNetwork::Id,
-             })
-        {
-            VerifyOrExit(callback(cmd, context) == Loop::Continue, /**/);
-        }
-    }
-    else if (mFeatureFlags.Has(Feature::kWiFiNetworkInterface))
+    bool hasNet  = mFeatureFlags.Has(Feature::kThreadNetworkInterface);
+    bool hasWifi = mFeatureFlags.Has(Feature::kWiFiNetworkInterface);
+    bool hasCred = mFeatureFlags.Has(Feature::kPerDeviceCredentials);
+    auto netId   = hasNet ? AddOrUpdateThreadNetwork::Id : AddOrUpdateWiFiNetwork::Id;
+
+    static constexpr size_t kNetworkCommands     = 5; // Count of Network Commands
+    static constexpr size_t kCredentialsCommands = 1; // Count of Credential Commands
+
+    static const DataModel::AcceptedCommandEntry commands[] = {
+        // A static array of the posible commands
+        { ScanNetworks::Id, {} },   //
+        { netId, {} },              //
```

Comment:
is this static given that netId is a variable that depends on a feature flag?

My instinct is that this is not static.